--- tf_savemodel_benchmark.py	2022-03-10 18:17:05.000625704 -0800
+++ tf_savemodel_benchmark_new.py	2022-03-10 18:22:27.696642636 -0800
@@ -146,6 +146,9 @@
         latency = avg_time * 1000
         throughput = 1.0 / avg_time * args.batch_size
         print("Latency: {:.0f} ms".format(latency))
+        import logging
+        logger = logging.getLogger('LOGGER_NAME')
+        logger.info('Throughput: %f', throughput)
         print("Throughput: {:.2f} fps".format(throughput))
 
 
