--- create_wds_shards.py	2022-09-20 14:55:48.732386718 -0700
+++ create_wds_shards_commonvoice.py	2022-09-20 14:53:56.554637629 -0700
@@ -27,7 +27,10 @@
     t, sr = torchaudio.load(audio_file_path)
 
     if sr != 16000:
-        raise ValueError("expected sampling rate of 16 kHz")
+        # Resample to 16kHz due to requirement
+        transform = torchaudio.transforms.Resample(orig_freq=sr, new_freq=16000)
+        t = transform(t)
+        #raise ValueError("expected sampling rate of 16 kHz")
 
     return t
 
@@ -61,27 +64,20 @@
     sample_keys_per_language = defaultdict(list)
 
     for f in audio_files:
-        # path should be
-        # voxlingua107_folder_path/<LANG_ID>/<VIDEO---0000.000-0000.000.wav>
-        m = re.match(
-            r"(.*/((.+)/.+---(\d\d\d\d\.\d\d\d)-(\d\d\d\d\.\d\d\d))\.wav)",
-            f.as_posix(),
-        )
+        # Common Voice format
+        # commonVoice_folder_path/common_voice_<LANG_ID>_00000000.wav'
+        m = re.match(r"((.*)(common_voice_)(.+)(_)(\d+).wav)", f.as_posix())
+
         if m:
             loc = m.group(1)
-            key = m.group(2)
-            lang = m.group(3)
-            start = float(m.group(4))
-            end = float(m.group(5))
-            dur = end - start
-            # Period is not allowed in a WebDataset key name
-            key = key.replace(".", "_")
-            if dur > min_dur:
-                # store statistics
-                all_language_ids.add(lang)
-                sample_keys_per_language[lang].append(key)
-                t = (key, lang, loc, dur)
-                data_tuples.append(t)
+            lang = m.group(4)
+            key = m.group(6)
+            dur = 0
+            # store statistics
+            all_language_ids.add(lang)
+            sample_keys_per_language[lang].append(key)
+            t = (key, lang, loc, dur)
+            data_tuples.append(t)
         else:
             raise Exception("Unexpected wav name: " + f)
 
