--- demo.py	2023-01-22 03:46:30.162956299 +0900
+++ demo-gpu.py	2023-01-22 03:53:57.998938230 +0900
@@ -36,7 +36,7 @@
     # Initialize the process group with ccl backend
     dist.init_process_group(backend='ccl')
 
-    device = 'cpu' #"xpu:{}".format(dist.get_rank())
+    device = "xpu:{}".format(dist.get_rank())
     model = Model().to(device)
     if dist.get_world_size() > 1:
         model = DDP(model, device_ids=[device] if (device != 'cpu') else None)
