--- text_classification_with_transformer.py	2022-09-20 02:24:42.814605146 -0700
+++ text_classification_with_transformer2.py	2022-09-20 02:24:48.489188611 -0700
@@ -27,6 +27,16 @@
 
 
 """
+## Bfloat16 mixed precision learning 
+"""
+
+from tensorflow.keras import mixed_precision
+
+policy = mixed_precision.Policy('mixed_bfloat16')
+mixed_precision.set_global_policy(policy)
+
+
+"""
 ## Implement a Transformer block as a layer
 """
 
