--- interfaces.py	2022-10-07 16:58:26.836359346 -0700
+++ interfaces_new.py	2022-10-07 16:59:09.968110128 -0700
@@ -945,7 +945,7 @@
         out_prob = self.mods.classifier(emb).squeeze(1)
         score, index = torch.max(out_prob, dim=-1)
         text_lab = self.hparams.label_encoder.decode_torch(index)
-        return out_prob, score, index, text_lab
+        return out_prob, score, index   # removed text_lab to get torchscript to work
 
     def classify_file(self, path):
         """Classifies the given audiofile into the given set of labels.
